syntax = "proto3";

package protocol.game;

enum E_PacketID
{
	MOVE_REQ = 0;
	PLANT_BOMB_REQ = 1;
	UPDATE_PLAYER_NOTIFY = 2;
	BOMB_EXPLOSION_NOTIFY = 3;
	WELCOME_NOTIFY = 4;
	GAME_START_NOTIFY = 5;
	GAME_OVER_NOTIFY = 6;
}
enum E_RespCode
{
	OK = 0;
}
enum E_Direction
{
	UP = 0;
	DOWN = 1;
	LEFT = 2;
	RIGHT = 3;
}


// not used for packet ( used for data struct )

message Vec2
{
	float x = 1;
	float y = 2;
}
message PlayerState
{
	int32 id = 1;
	Vec2 pos = 2;
}
//


message Move_Req
{
	int32 seqID = 1;
	E_Direction dir = 2;
}
message BombPlant_Notify
{
	Vec2 pos = 1;
	int32 power = 2;
}
message BombExplosion_Notify
{
	repeated Vec2 positions = 1;
}
message Welcome_Notify
{
	int32 id = 1;
	repeated PlayerState players = 2;
}
message Update_Notify
{
	int32 lpi = 1;
	repeated PlayerState players = 2;
}
message GameOver_Notify
{
	bool isWin = 1;
}